mono/metadata/sre.c

```diff
@@ -2831,9 +2832,8 @@ reflection_methodbuilder_to_mono_method (MonoClass *klass,
 		if (image_is_dynamic (klass->image))
 			g_hash_table_insert (((MonoDynamicImage*)klass->image)->method_aux_hash, m, method_aux);
 
-		mono_loader_unlock ();
+		goto leave;
```

Comment:
When there is some common cleanup in function exit paths, it is imho really really preferable to avoid "early return". (or rely on C++ destructors in the hypothetical and distant future). This is even more so if/when this code uses handles -- but those are just an example that suggests the rule -- do not repeat common exit code, whether it is more easy or less easy.